src/test/java/BaseTest.java

```diff
@@ -0,0 +1,106 @@
+
+        htmlReporter = new ExtentSparkReporter(System.getProperty("user.dir") + File.separator + "reports"
+                + File.separator + "AutomationReport.html");
+        htmlReporter.config().setEncoding("utf-8");
+        htmlReporter.config().setDocumentTitle("Automation Report");
+        htmlReporter.config().setReportName("Automation Test Results");
+        htmlReporter.config().setTheme(Theme.DARK);
+        extent = new ExtentReports();
+        extent.attachReporter(htmlReporter);
+        extent.setSystemInfo("Automation Tester", "Balazs Kocsis");
+    }
+
+    @BeforeMethod
+    @Parameters(value = {"browserName"})
+    public void beforeMethod(String browserName, Method testMethod) {
+
+        logger = extent.createTest(testMethod.getName());
+        setupDriver(browserName);
+        driver.manage().window().maximize();
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        setupEnvironment();
+    }
+
+    @AfterMethod
+    public void afterMethod(ITestResult result) {
+
+        if (result.getStatus() == ITestResult.SUCCESS) {
+            String methodName = result.getMethod().getMethodName();
+            String logText = "Test Case: " + methodName + " Passed";
+            Markup m = MarkupHelper.createLabel(logText, ExtentColor.GREEN);
+            logger.log(Status.PASS, m);
+        } else if (result.getStatus() == ITestResult.FAILURE) {
+            String methodName = result.getMethod().getMethodName();
+            String logText = "Test Case: " + methodName + " Failed";
+            Markup m = MarkupHelper.createLabel(logText, ExtentColor.RED);
+            logger.log(Status.FAIL, m);
+        }  else if (result.getStatus() == ITestResult.SKIP) {
+            String methodName = result.getMethod().getMethodName();
+            String logText = "Test Case: " + methodName + " Skipped";
+            Markup m = MarkupHelper.createLabel(logText, ExtentColor.YELLOW);
+            logger.log(Status.SKIP, m);
+        }
+        driver.quit();
+    }
+
+    @AfterTest
+    public void afterTest() {
+        extent.flush();
+    }
+
+    public void setupDriver(String browserName) {
+
+        if (browserName.equalsIgnoreCase("chrome")) {
+            System.setProperty("webdriver.chrome.driver", System.getProperty("user.dir") + File.separator + "drivers" +
+                    File.separator + "chromedriver.exe");
+            driver = new ChromeDriver();
+        } else if (browserName.equalsIgnoreCase("firefox")) {
+            System.setProperty("webdriver.gecko.driver", System.getProperty("user.dir") + File.separator + "drivers" +
+                    File.separator + "geckodriver.exe");
+            driver = new FirefoxDriver();
+        } else {
+            System.setProperty("webdriver.chrome.driver", System.getProperty("user.dir") + File.separator + "drivers" +
+                    File.separator + "chromedriver.exe");
+            driver = new ChromeDriver();
```

Comment:
Please add some debug console input to identify that fallback back option has been chosen.
It's important to avoid confusion when options for `Firefox` was selected, but with extra space of minor typo which will lead to silent execution tests on `Chrome`.